MetaMask/metamask-mobile
# feat: Enable token detection on arbitrum, optimism, base, and zksync

## **Description**

Enables token detection on arbitrum, optimism, base, and zksync.

Patching these from core:
- https://github.com/MetaMask/core/pull/2035
- https://github.com/MetaMask/core/pull/3584

## **Related issues**

## **Manual testing steps**

1. Restore an account with tokens on arbitrum, optimism, base, and zksync.
2. enable token detection in settings
3. Switch to each of those chains
4. It should suggest <n> new token(s) found in this account
5. When manually importing a token, there should be a "search" tab

## **Screenshots/Recordings**



### **Before**

https://github.com/MetaMask/metamask-mobile/assets/3500406/a75f0412-31df-48ff-9832-262ecac84361

### **After**

https://github.com/MetaMask/metamask-mobile/assets/3500406/c6dc7170-1544-4d48-9709-c1952ba9d830

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
